packages/manager/modules/nutanix/src/dashboard/component/node-action-menu/controller.js

```diff
@@ -0,0 +1,32 @@
+import { POSSIBLE_ACTIONS } from './constants';
+import { MIN_NODES_BY_CLUSTER } from '../../../constants';
+
+export default class NutanixDashboardNodeActionMenu {
+  /* ngInject */
+  constructor($translate) {
+    this.$translate = $translate;
+  }
+
+  $onInit() {
+    this.mapPossibleActions();
+    this.POSSIBLE_ACTIONS = POSSIBLE_ACTIONS;
+  }
+
+  mapPossibleActions() {
+    this.possibleActionsMapped = {};
+    this.possibleActions.forEach(({ action, isPossible, reason }) => {
+      this.possibleActionsMapped[action] = { isPossible, reason };
+    });
+  }
+
+  getReasonTranslation(reason) {
+    if (!reason || reason.lenght === 0) return '';
```

Comment:
typo, btw can be reduced
```suggestion
    if (!reason?.length) return '';
```